src/coins.h

```diff
@@ -257,9 +274,24 @@ struct CoinsViewCacheCursor
     inline CoinsCachePair* Begin() const noexcept { return m_sentinel.second.Next(); }
     inline CoinsCachePair* End() const noexcept { return &m_sentinel; }
 
-    inline CoinsCachePair* NextAndMaybeErase(CoinsCachePair& current) noexcept { return current.second.Next(); }
+    //! Return the next entry after current, erasing current if WillErase(current) returns true
```

Comment:
In 6453d8e38a51311b23c980c5c61a60a0d0bd68b2 "coins: move Sync logic to CoinsViewCacheCursor"

This comment seems to be incorrect, `WillErase(current)` will return true when `m_will_erase == true`, but this function will not erase the entry in that instance.

Reply:
Good catch @achow101, updated the comment.